src/merkle_proof/proof.rs

```diff
@@ -23,7 +24,14 @@ pub fn merkle_opening_gadget<H>(
     H: kelvin::ByteHash,
 {
     // Generate a `PoseidonBranch` from the kelvin Branch.
-    let branch = PoseidonBranch::from(&branch);
+    let mut branch = PoseidonBranch::from(&branch);
+
+    let n_extensions = branch.extend(17);
```

Comment:
Same as avobe. Should not be hardcoded. We should compute the difference between the levels we have, and the levels we require to extend which is provided by the user of the lib.

Reply:
fixed